guides/source/active_model_basics.md

```diff
@@ -285,101 +800,308 @@ irb> person = Person.new
+
+Currently, when including `ActiveModel::Model` you get the features from
+[ActiveModel::API](active_model_basics.html#api).
 
 ### Serialization
 
-`ActiveModel::Serialization` provides basic serialization for your object.
-You need to declare an attributes Hash which contains the attributes you want to
-serialize. Attributes must be strings, not symbols.
+`ActiveModel::Serialization` provides basic serialization to a serializable hash
+for your objects. You need to declare an attributes Hash which should contain
+the attributes you want to serialize. Attributes must be strings, not symbols.
 
 ```ruby
 class Person
   include ActiveModel::Serialization
 
-  attr_accessor :name
+  attr_accessor :name, :age
 
   def attributes
-    { 'name' => nil }
+    # compulsory declaration of attributes to serialize
+    {"name" => nil, "age" => nil}
+  end
+
+  def capitalized_name
+  # an example of how we can define a method that will be included a serialized hash
+    name.capitalize
   end
 end
 ```
 
-Now you can access a serialized Hash of your object using the `serializable_hash` method.
+Now you can access a serialized Hash of your object using the
+`serializable_hash` method. Valid options for serializable hash include `:only`,
+`:except`, `:methods` and `:include`.
 
 ```irb
 irb> person = Person.new
+
 irb> person.serializable_hash
-=> {"name"=>nil}
-irb> person.name = "Bob"
+=> {"name"=>nil, "age"=>nil}
+
+# setting the name and age attributes and then serializing the object
+irb> person.name = "bob"
+irb> person.age = 22
 irb> person.serializable_hash
-=> {"name"=>"Bob"}
+=> {"name"=>"bob", "age"=>22}
+
+# using the methods option to include the capitalized_name method
+irb>  person.serializable_hash(methods: :capitalized_name)
+=> {"name"=>"bob", "age"=>22, "capitalized_name"=>"Bob"}
+
+# using the only method to include only the name attribute
+irb> person.serializable_hash(only: :name)
+=> {"name"=>"bob"}
+
+# using the except method to exclude the name attribute
+irb> person.serializable_hash(except: :name)
+=> {"age"=>22}
+```
+
+The example to utilize the `includes` option requires a slightly more complex
+scenario as defined below:
+
+```ruby
+  class Person
+   include ActiveModel::Serialization
+   attr_accessor :name, :notes # Emulate has_many :notes
+
+    def attributes
+       {"name" => nil}
+    end
+  end
+
+  class Note
+    include ActiveModel::Serializers::JSON
+    attr_accessor :title, :text
+    def attributes
+      {"title" => nil, "text" => nil}
+    end
+  end
+```
+
+
+```irb
+irb> note = Note.new
+irb> note.title = "Weekend Plans"
+irb> note.text = "Some text here"
+
+irb> person = Person.new
+irb> person.name = "Napoleon"
+irb> person.notes = [note]
+
+irb> person.serializable_hash
+=> {"name" => "Napoleon"}
+
+irb> person.serializable_hash(include: { notes: { only: "title" }})
+=> {"name" => "Napoleon", "notes" => [{"title"=>"Weekend Plans"}]}
 ```
 
 #### ActiveModel::Serializers
```

Comment:
I think this title can be : #### ActiveModel::Serializers::JSON and the one on line 1097 can be removed.

Reply:
I think the line numbers have changed since you commented, can you point out which header you were referring to on line 1097?

Reply:
I think we can change the main header here from ActiveModel::Serializers to ActiveModel::Serializers::JSON since this whole section refers to JSON serializing.

<img width="714" alt="Screenshot 2024-02-23 at 14 10 01" src="https://github.com/Ridhwana/rails/assets/58528404/a687fac9-b2e4-47c1-a4c4-d320048d773d">